Userland/Services/WindowServer/ClientConnection.cpp

```diff
@@ -1129,7 +1129,11 @@ void ClientConnection::set_window_parent_from_client(i32 client_id, i32 parent_i
     if (!parent_window)
         did_misbehave("SetWindowParentFromClient: Bad parent window ID");
 
-    child_window->set_parent_window(*parent_window);
+    if (parent_window->is_stealable()) {
```

Comment:
This still seems too lax. Any application can still steal the window, as long as it's marked as stealable.

Can we arrange things so that only the intended window thief can get to it?

Reply:
This is making me think about multiple window thieves, I'd like to save that for later and just store `Optional<i32> m_thief_client_id` in `Window` for now.